AWSIoT/AWSIoTDataManager.m

```diff
@@ -333,6 +333,17 @@ - (void)enableMetricsCollection:(BOOL)enabled {
     [self.mqttClient setIsMetricsEnabled:enabled];
 }
 
+- (void)addUserMetaData:(NSDictionary<NSString *, NSString *> *)userMetaData {
+    [self.mqttClient setUserMetaData:userMetaData];
+
+    // validate the length of username field
+    NSMutableString *userMetadata = [self.mqttClient getUserMetaData];
+    NSString *baseUsername = [NSString stringWithFormat:@"%@%@", @"?SDK=iOS&Version=", [self.mqttClient SDK_VERSION]];
+    if ([userMetadata length] > 255) {
```

Comment:
Throwing a NSException can crash the calling app if used from Swift. Instead, lets log this as an error and truncate the data so that it fits.

Reply:
Done